website/payments/admin.py

```diff
@@ -190,6 +245,53 @@ def process_wire_selected(self, request: HttpRequest, queryset: QuerySet) -> Non
 
     process_wire_selected.short_description = _("Process selected payments (wire)")
 
+    def add_to_new_batch(self, request: HttpRequest, queryset: QuerySet) -> None:
+        """Add selected TPAY payments to a new batch"""
+        tpays = queryset.filter(type=Payment.TPAY)
+        if len(tpays) > 0:
+            batch = Batch()
+            batch.save()
+            tpays.update(batch=batch)
+        _show_message(
+            self,
+            request,
+            len(tpays),
+            f"Successfully added {len(tpays)} payments to new batch",
+            f"No payments using Thalia Pay are selected, " f"no batch is created",
+        )
+
+    add_to_new_batch.short_descriptionn = _("Add selected TPAY payments to a new batch")
+
+    def add_to_last_batch(self, request: HttpRequest, queryset: QuerySet) -> None:
+        """Add selected TPAY payments to the last batch"""
+        tpays = queryset.filter(type=Payment.TPAY)
+        if len(tpays) > 0:
+            batch = Batch.objects.last()
+            if not batch:
+                self.message_user(request, "No batches available.", messages.ERROR)
+            elif not batch.processed:
+                batch.save()
+                tpays.update(batch=batch)
+                self.message_user(
+                    request,
+                    f"Successfully added {len(tpays)} payments to {batch}",
+                    messages.SUCCESS,
+                )
+            else:
+                self.message_user(
+                    request,
+                    f"The last batch {batch} is already processed",
+                    messages.ERROR,
+                )
+        else:
+            self.message_user(
+                request,
+                f"No payments using Thalia Pay are selected, " f"no batch is created",
+                messages.ERROR,
+            )
```

Comment:
This whole part is doing business logic in admin.py, this should happen in services.py instead.
Check the `process_card_selected` function for an example, the result of your function could be the message and the status.

Reply:
Same here, it is not really possible to put the messages in services and the only common thing in these functions is `tpays.update(batch=batch)` as far as I can see.

Reply:
The function would be all the if and else statements with the messages as string and level outputs. The output can then be used here to show a message based on what happened